### ISSUE TITLE
PHP: Go async with AMP

### ISSUE THREAD

**Issue by lennartdohmann** [GDATASoftwareAG/vaas#451]
Use [AMPHP](https://github.com/amphp/amp?tab=readme-ov-file) to go async

**Comment by unglaublicherdude** [comment]
I tested the implementation on an IONOS webhosting (shared-hosting) server and it does run there. :+1:

**Comment by unglaublicherdude** [comment]
We ran into an issue with async php by using the revolt/eventloop (dependency of the amphp packgages).

The nextcloud-app fails (only in the ci) [when updating the http-client to the amphp one](https://github.com/GDATASoftwareAG/nextcloud-gdata-antivirus/pull/98).

There is an issue with the revolt/eventloop and opcache which we can workaround by deactivate the opcache for the vendor/amphp directory. But for now, we decided to don't put anymore effort into this. 

I opened this https://github.com/revoltphp/event-loop/issues/95.
I opened this https://github.com/amphp/http-client/issues/365.

With this issue, [this PR is also basically obsolete too](https://github.com/GDATASoftwareAG/vaas/pull/544).

What we still have to do is to revert the http-client to the guzzle one or to build a factory that creates an http-client regarding to some ini-settings (opcache.enabled would lead to good old guzzle).
